client/session.go

```diff
@@ -717,17 +721,20 @@ func (s *session) FetchAll(namespace string, ids []string, startInclusive, endEx
 	return iters, nil
 }
 
-func (s *session) consistencyResult(majority, enqueued, resultErrs int32, resultErr error) error {
+func (s *session) consistencyResult(
+	majority, enqueued, responded, resultErrs int32,
+	errors []error,
+) error {
 	if resultErrs == 0 {
 		return nil
 	}
 
 	// Check consistency level satisfied
 	success := enqueued - resultErrs
 	reportErr := func() error {
-		return xerrors.NewRenamedError(resultErr, fmt.Errorf(
-			"failed to meet %s with %d/%d success, first error: %v",
-			s.level.String(), success, enqueued, resultErr))
+		return xerrors.NewRenamedError(errors[0], fmt.Errorf(
```

Comment:
Correct, this is a little bit of a hack to give somewhat usefulness to the `IsBadRequestError` or `IsInternalServerError` functions that act on the underlying error underneath the renamed error.